include/librealsense2/dds/dds-topic-reader.cpp

```diff
@@ -0,0 +1,109 @@
+    history().depth = 10;  // default is 1
+
+    // We don't want to miss connection/disconnection events
+    // (default is best-effort)
+    reliability().kind = eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS;
+
+    // The Subscriber receives samples from the moment it comes online, not before
+    // (this is the default, but bears repeating)
+    durability().kind = eprosima::fastdds::dds::VOLATILE_DURABILITY_QOS;
+
+    // The writer-reader handshake is always done on UDP, even when data_sharing (shared memory) is used for
+    // actual messaging. This means it is possible to send a message and receive it on the reader's
+    // side even before the UDP handshake is complete:
+    // 
+    //   1. The writer goes up and broadcasts its presence periodically; no readers exist
+    //   2. The reader joins and broadcasts its presence, again periodically; it doesn<92>t know about the writer yet
+    //   3. The writer sees the reader (in-between broadcasts) so sends a message
+    //   4. The reader gets the message and discards it because it does not yet recognize the writer
+    // 
+    // This depends on timing. When shared memory is on, step 3 is so fast that this miscommunication is much more
+    // likely. This is a known gap in the DDS standard.
+    //
+    // We can either insert a sleep between writer creation and message sending or we can disable
+    // data_sharing for this topic, which we did here.
+    // 
+    // See https://github.com/eProsima/Fast-DDS/issues/2641
+    //
+    data_sharing().off();
+}
+
+
+void dds_topic_reader::run( reader_qos const & rqos )
+{
+    // The Subscriber manages the activities of several DataReader entities
+    _subscriber = DDS_API_CALL(
+        _topic->get_participant()->get()->create_subscriber( eprosima::fastdds::dds::SUBSCRIBER_QOS_DEFAULT ) );
+
+    // The DataReader is the Entity used by the application to subscribe to updated values of the data
+    eprosima::fastdds::dds::StatusMask status_mask;
+    if( _on_subscription_matched )
+        status_mask << eprosima::fastdds::dds::StatusMask::subscription_matched();
+    if( _on_data_available )
+        status_mask << eprosima::fastdds::dds::StatusMask::data_available();
+    _reader = DDS_API_CALL(
+        _subscriber->create_datareader( _topic->get(), rqos, status_mask.any() ? this : nullptr, status_mask ) );
+}
+
+
+void dds_topic_reader::on_subscription_matched(
+    eprosima::fastdds::dds::DataReader *, eprosima::fastdds::dds::SubscriptionMatchedStatus const & info )
+{
+    // Called when the subscriber is matched (un)with a Writer
+    if( _on_subscription_matched )
+        _on_subscription_matched( info );
+}
+
+void dds_topic_reader::on_data_available( eprosima::fastdds::dds::DataReader * )
+{
+    // Called when a new  Data Message is received
+    if( _on_data_available )
+        _on_data_available();
+}
+
+
+}  // namespace dds
+}  // namespace librealsense
```

Comment:
Missing EOL. GitHub shows an "missing EOL" error indication for this file

Reply:
@OhadMeir you are getting into trouble ... ;-)